### PR TITLE
feat(replays): added decompression for replay attachments

### LINKED ISSUE #35150: Decompress replay data within Replay Details
Compressed payloads uploaded by the SDK need to be decompressed in order for us to display them in the replay details.

### PR DESCRIPTION
**Pull request by Danondso**
Closes #35150 
Installed pako so we can decompress rrweb attachments for replays.

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: static/app/utils/replays/hooks/useReplayData.tsx [hunk 1]
```diff
@@ -115,9 +140,25 @@ function useReplayData({eventSlug, orgId}: Options): Result {
...
+            includeAllArgs: true,
+          }
         );
-        return JSON.parse(response) as ReplayAttachment;
+
+        // for non-compressed events, parse and return
+        try {
+          return JSON.parse(response[0]) as ReplayAttachment;
+        } catch (error) {
+          // swallow exception as if we can't parse it, it's going to be compressed
```
Comment (Danondso) [inline static/app/utils/replays/hooks/useReplayData.tsx:153]:
```suggestion
          // swallow exception.. if we can't parse it, it's going to be compressed
```

File: static/app/utils/replays/hooks/useReplayData.tsx [hunk 2]
```diff
@@ -67,6 +68,30 @@ const IS_RRWEB_ATTACHMENT_FILENAME = /rrweb-[0-9]{13}.json/;
...
+    breadcrumbs: [],
+    replaySpans: [],
+    recording: [],
+  };
+
+  replayAttachments.recording = unsortedReplayAttachments;
+  replayAttachments.recording.forEach(record => {
+    if (!Number.isInteger(record.timestamp)) {
+      record.timestamp = record.timestamp * 1000;
+    }
```
Comment (billyvg) [inline static/app/utils/replays/hooks/useReplayData.tsx:82]: I'm going to make sure the SDK sends it in ms so this won't be necessary.

**Review by billyvg — COMMENTED** [review]
Nice, I think this is good to go, will wait to merge until SDK is ready (there might be some updates needed here if we do change some SDK details though).

File: static/app/utils/replays/hooks/useReplayData.tsx [hunk 3]
```diff
@@ -115,9 +135,30 @@ function useReplayData({eventSlug, orgId}: Options): Result {
     const attachments = await Promise.all(
       rrwebAttachmentIds.map(async attachment => {
         const response = await api.requestPromise(
-          `/api/0/projects/${orgId}/${projectId}/events/${eventId}/attachments/${attachment.id}/?download`
+          `/api/0/projects/${orgId}/${projectId}/events/${eventId}/attachments/${attachment.id}/?download`,
```
[the remark this answers is not in the document]
Comment (billyvg) [reply static/app/utils/replays/hooks/useReplayData.tsx:138]: It gets caught when we call it.

**Comment by billyvg** [comment]
@Danondso I'm gonna go ahead and merge since this is backwards compatible, we can follow-up with any SDK changes we make later on.

**Review by billyvg — APPROVED** [review]

**Pull request merged**